### PR TITLE
feat: add multi-format plugin manifest generator (Codex + Antigravity)

### PR DESCRIPTION
**Pull request by amondnet**
## Summary

Adds a `multi-format` manifest generator (`scripts/multi-format.ts`) that produces both **Codex** and **Antigravity** plugin manifests for all 53 local plugins, and a Codex marketplace index. This makes the plugin collection usable out of the box across three runtimes: Claude Code, Codex, and Antigravity.

## New Files and Commands

| Path | Purpose |
|---|---|
| `scripts/multi-format.ts` | Generator — reads each plugin's Claude Code `plugin.json` and emits runtime-specific manifests |
| `scripts/multi-format.test.ts` | 25 unit tests covering generation logic, edge cases, and output shape |
| `scripts/cli.ts` (updated) | Added `multi-format` sub-command |
| `package.json` (updated) | Added `plugins:multi-format` npm/bun script |
| `.agents/plugins/marketplace.json` | Codex marketplace index for all 53 plugins |
| `CLAUDE.md` (updated) | Documents the multi-runtime manifest section |

**Per-plugin generated artifacts (×53 plugins):**

- `.codex-plugin/plugin.json` — Codex manifest
- `.mcp.json` — Codex MCP server config (plugins with MCP servers)
- `plugin.json` (root) — Antigravity manifest
- `mcp_config.json` — Antigravity MCP server config (plugins with MCP servers)
- `hooks.json` — Antigravity hooks config (plugins with hooks)

## Usage

```bash
# Generate manifests for all local plugins
bun scripts/cli.ts multi-format

# Or via npm script
bun run plugins:multi-format
```

## Validation

All 53 plugins pass both validators:

```bash
# Claude Code validator
claude plugin validate plugins/<plugin-name>

# Codex validator
python validate_plugin.py plugins/<plugin-name>/.codex-plugin/plugin.json
```

## Test Plan

- [ ] Run `bun scripts/multi-format.test.ts` — all 25 tests should pass
- [ ] Run `bun scripts/cli.ts multi-format` — generates manifests without errors
- [ ] Spot-check a plugin with MCP: `claude plugin validate plugins/fetch`
- [ ] Spot-check a plugin without MCP: `claude plugin validate plugins/gatekeeper`

## Notes

- Version and changelog are managed by release-please; no manual edits made to `CHANGELOG.md` or plugin `version` fields.
- Generated files in `plugins/*/plugin.json`, `plugins/*/.codex-plugin/plugin.json`, etc. are committed to the repo so users can install without running the generator locally.

---
## Summary by cubic
Adds a multi-format manifest generator and pre-generates Codex + Antigravity manifests for all 53 local plugins, plus a Codex marketplace index. Hardens error handling and validation so malformed or failed plugins stop the run and surface in CI.

- **New Features**
  - Added `scripts/multi-format.ts`, `scripts/cli.ts multi-format`, and `package.json` script `plugins:multi-format`.
  - Converts Claude `plugin.json` into Codex `.codex-plugin/plugin.json` (+ `./.mcp.json`) and Antigravity `plugin.json` (+ `mcp_config.json`, `hooks.json`); generates `.agents/plugins/marketplace.json`.
  - Rules and docs: synthesize Codex `interface`, externalize MCP configs, mirror hooks, normalize `skills` path. Updated `CLAUDE.md`. All 53 plugins pass Claude and Codex validators.
  - Source of truth stays the Claude manifest; regenerate via `bun scripts/cli.ts multi-format` or `bun run plugins:multi-format`.

- **Bug Fixes**
  - `readClaudeManifest` now throws on malformed JSON instead of treating it as missing; no more silent skips.
  - CLI wraps per-plugin generation, reports failures, and exits non-zero if any plugin fails.
  - Privacy and correctness: strip `author.email` from generated manifests; handle empty/array `mcpServers`, delete stale MCP config files, dedupe Codex marketplace by name, and avoid stale `mcpServers` fields in Codex manifests. Tests increased to 44.

<sup>Written for commit 0864c64a915ddb5981a32b71972d900f2dade832. Summary will update on new commits.</sup>

### CODE REVIEW

File: scripts/multi-format.ts [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+  developerName: string
+  category: string
+  capabilities: string[]
+  defaultPrompt: string[]
+  websiteURL?: string
+}
+
+export interface CodexAuthor {
+  name: string
+  email?: string
```
[the remark this answers is not in the document]
Comment (amondnet) [reply scripts/multi-format.ts:77]: 💡 Implemented with alternative approach ([0864c64](https://github.com/pleaseai/claude-code-plugins/pull/185/commits/0864c64a915ddb5981a32b71972d900f2dade832))

- Suggestion: Strip email and extract into a shared helper.
- Applied: New `sanitiseAuthor()` helper drops `email` and re-validates the https URL; both `pickAuthor()` (Codex path) and `toAntigravityManifest()` go through it now.
- Reason: A single helper keeps the privacy rule in one place and lets the call sites just say 'sanitise this author'.

Thanks for the suggestion, @gemini-code-assist!

File: scripts/multi-format.ts [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+function normalizeVersion(value: string | undefined): string {
+  if (!value) return "1.0.0"
+  return /^\d+\.\d+\.\d+(?:[-+].+)?$/.test(value) ? value : "1.0.0"
+}
+
+function pickAuthor(claude: ClaudePluginManifest): CodexAuthor {
+  const src = claude.author
+  if (!src?.name) return { name: "Community" }
+  const author: CodexAuthor = { name: src.name }
+  if (src.email) author.email = src.email
```
[the remark this answers is not in the document]
Comment (amondnet) [reply scripts/multi-format.ts]: ✅ Applied in [0864c64](https://github.com/pleaseai/claude-code-plugins/pull/185/commits/0864c64a915ddb5981a32b71972d900f2dade832)

- File: `scripts/multi-format.ts`
- Change: `pickAuthor()` (Codex path) now routes through the shared `sanitiseAuthor()` helper, which strips `email` before propagating. claude-md-management / worktree manifests no longer carry the contributor email.

Thanks for the suggestion, @gemini-code-assist!

File: scripts/multi-format.ts [hunk 3]
```diff
@@ -0,0 +1,426 @@
...
+ *
+ * Antigravity only requires `name`; other fields are preserved when present
+ * so the same file can satisfy both Antigravity and the legacy root-level
+ * Claude Code manifest convention used by a few existing plugins.
+ */
+export function toAntigravityManifest(claude: ClaudePluginManifest): AntigravityManifest {
+  const manifest: AntigravityManifest = { name: claude.name }
+  if (claude.version) manifest.version = claude.version
+  if (claude.description) manifest.description = claude.description
+  if (claude.author) manifest.author = claude.author
```
[the remark this answers is not in the document]
Comment (amondnet) [reply scripts/multi-format.ts]: ✅ Applied in [0864c64](https://github.com/pleaseai/claude-code-plugins/pull/185/commits/0864c64a915ddb5981a32b71972d900f2dade832)

- File: `scripts/multi-format.ts`
- Change: `toAntigravityManifest()` now goes through `sanitiseAuthor()` instead of passing `claude.author` through verbatim, so the email is dropped on the Antigravity side too.

Thanks for the suggestion, @gemini-code-assist!

File: scripts/multi-format.ts [hunk 4]
```diff
@@ -0,0 +1,426 @@
...
+  return manifest
+}
+
+/**
+ * Pull inline `mcpServers` out of a Claude manifest so it can be written to
+ * an external file (`.mcp.json` for Codex, `mcp_config.json` for Antigravity).
+ * Returns null when the manifest has no inline servers.
+ */
+export function extractMcpServersFile(claude: ClaudePluginManifest): { mcpServers: Record<string, unknown> } | null {
+  if (!claude.mcpServers || typeof claude.mcpServers !== "object") return null
```
[the remark this answers is not in the document]
Comment (amondnet) [reply scripts/multi-format.ts]: ✅ Applied in [0864c64](https://github.com/pleaseai/claude-code-plugins/pull/185/commits/0864c64a915ddb5981a32b71972d900f2dade832)

- File: `scripts/multi-format.ts`
- Change: `extractMcpServersFile()` now returns null for empty `{}` mcpServers (also added an Array.isArray guard). The generator also deletes any stale `.mcp.json` / `mcp_config.json` from previous runs so they don't linger. Removed the two empty companion files that were generated for `plugins/google-workspace/`.

Thanks for catching this, @gemini-code-assist!

File: scripts/multi-format.ts [hunk 5]
```diff
@@ -0,0 +1,426 @@
...
+  name?: string
+  plugins: Array<MarketplaceEntry & { source?: unknown }>
+}
+
+export function toCodexMarketplace(claudeMarketplace: ClaudeMarketplace): {
+  name: string
+  interface: { displayName: string }
+  plugins: CodexMarketplaceEntry[]
+} {
+  const localPlugins = claudeMarketplace.plugins.filter(p => typeof p.source === "string" && p.source.startsWith("./plugins/"))
```
[the remark this answers is not in the document]
Comment (amondnet) [reply scripts/multi-format.ts]: ✅ Applied in [0864c64](https://github.com/pleaseai/claude-code-plugins/pull/185/commits/0864c64a915ddb5981a32b71972d900f2dade832)

- File: `scripts/multi-format.ts`
- Change: `toCodexMarketplace()` now deduplicates by plugin name (keeps first occurrence) using a Set, exactly as suggested. There are no duplicates in the current marketplace.json, but this is cheap defence against accidental future double-entries.

Thanks for the suggestion, @gemini-code-assist!
